specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2018-10-01/imds.json

```diff
@@ -0,0 +1,731 @@
+                    },
+                    {
+                        "name": "Metadata",
+                        "in": "header",
+                        "type": "string",
+                        "enum": [
+                            "true"
+                        ],
+                        "description": "This must be set to 'true'.",
+                        "required": true
+                    }   
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/instance"
+                        }
+                    },
+                    "400": {
+                        "description": "Invalid request",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    },
+                    "403": {
+                        "description": "Forbidden",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    },
+                    "404": {
+                        "description": "Not found",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    },
+                    "405": {
+                        "description": "Method not allowed",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    },
+                    "429": {
+                        "description": "Too many requests",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    },
+                    "503": {
+                        "description": "Service unavailable",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    },
+                    "500": {
+                        "description": "Server error",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "Get Instance Metadata for the VM": {
+                        "$ref": "./examples/getinstancemetadata.json"
+                    }
+                }
+            }
+        },
+        "/attested/document": {
+            "get": {
+                "operationId": "getAttested",
+                "description": "Get Attested Data for the Virtual Machine.",
+                "parameters": [
+                    {
+                        "name": "api-version",
```

Comment:
Usually the api-version is defined as a global parameter (top level parameter property in the swagger spec). It has a special meaning to it. All the parameters specified in the the global parameters section will end up as properties on the generated client, unless specified to not do so via ["x-ms-parameter-location"](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-parameter-location) extension.

Reply:
Moved api-version to "parameters" section, under "ApiVersionParameter"